bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonDecoder.kt

```diff
@@ -176,7 +181,7 @@ internal open class DefaultBsonDecoder(
         return reader.state != AbstractBsonReader.State.END_OF_DOCUMENT && reader.currentBsonType != BsonType.NULL
     }
 
-    override fun decodeObjectId(): ObjectId = readOrThrow({ reader.readObjectId() }, BsonType.OBJECT_ID)
+    override fun decodeObjectId(): ObjectId = reader.readObjectId()
```

Comment:
Lets keep the `readOrThrow({ reader.readObjectId() }, BsonType.OBJECT_ID)` so that users know if an invalid type is present for the objectId field.